PhpAmqpLib/Wire/IO/SocketIO.php

```diff
@@ -34,14 +36,15 @@ public function __construct(
         ?AMQPConnectionConfig $config = null
     ) {
         $this->config = $config;
-        $this->host = $host;
+        $this->host = str_replace(['[', ']'], '', $host);
         $this->port = $port;
         $this->read_timeout = (float)$read_timeout;
         $this->write_timeout = (float)($write_timeout ?: $read_timeout);
         $this->heartbeat = $heartbeat;
         $this->initial_heartbeat = $heartbeat;
         $this->keepalive = $keepalive;
         $this->canDispatchPcntlSignal = $this->isPcntlSignalEnabled();
+        $this->isIpv6 = $this->isIpv6();
```

Comment:
We don't need to keep this variable as class property if it is being used in only one place and only once.